src/entries/popup/components/CommandK/useSearchableContacts.ts

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useMemo } from 'react';
+import { Address } from 'wagmi';
+
+import { i18n } from '~/core/languages';
+import { useContactsStore } from '~/core/state/contacts';
+import { useFeatureFlagsStore } from '~/core/state/currentSettings/featureFlags';
+import { KeychainType } from '~/core/types/keychainTypes';
+import { truncateAddress } from '~/core/utils/address';
+import { POPUP_URL, goToNewTab } from '~/core/utils/tabs';
+import { triggerAlert } from '~/design-system/components/Alert/Alert';
+
+import { useContacts } from '../../hooks/useContacts';
+import { useCurrentWalletTypeAndVendor } from '../../hooks/useCurrentWalletType';
+import { useIsFullScreen } from '../../hooks/useIsFullScreen';
+import { useRainbowNavigate } from '../../hooks/useRainbowNavigate';
+import { useWallets } from '../../hooks/useWallets';
+import { ROUTES } from '../../urls';
+
+import { ContactSearchItem, SearchItemType } from './SearchItems';
+import { PAGES } from './pageConfig';
+import { actionLabels } from './references';
+import { truncateName } from './useSearchableWallets';
+
+interface UseSearchableContactsParameters {
+  showLabel: boolean;
+}
+
+export const useSearchableContacts = ({
+  showLabel,
+}: UseSearchableContactsParameters) => {
+  const contacts = useContacts();
+  const { setSelectedContactAddress } = useContactsStore();
+  const { isWatchingWallet } = useWallets();
+  const { featureFlags } = useFeatureFlagsStore();
+  const navigate = useRainbowNavigate();
+
+  const isFullScreen = useIsFullScreen();
+
+  const { type, vendor } = useCurrentWalletTypeAndVendor();
+
+  const isTrezor = useMemo(() => {
+    return type === KeychainType.HardwareWalletKeychain && vendor === 'Trezor';
+  }, [type, vendor]);
+
+  const allowSend = useMemo(
+    () => !isWatchingWallet || featureFlags.full_watching_wallets,
+    [featureFlags.full_watching_wallets, isWatchingWallet],
+  );
+
+  const shouldNavigateToSend = useMemo(() => {
+    // Trezor should always be in a new tab
+    return !(isTrezor && !isFullScreen) && allowSend;
+  }, [allowSend, isFullScreen, isTrezor]);
```

Comment:
is there a reason to memoize these primitive values?

Reply:
Yeah i just copied from [here](https://github.com/rainbow-me/browser-extension/blob/1f33946170e69e71344a18246da3ab6f9f95455e/src/entries/popup/pages/home/Header.tsx#L160-L163)

Reply:
@magiziz outside of copying code from another area, is there a reason to memoize these values?

Reply:
Nope i don't think we need them to be memoized since it's just returning a boolean and booleans shouldn't be memoized. Pushed a fix for this 👍